Task.1.2.Matrix/Task.1.2.Matrix/Entities/SquareMatrix.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task1.Two.Matrix.Entities
+{
+    /// <summary>
+    /// A generic class to create data structure in Square matrix
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal class SquareMatrix<T>
+    {
+        /// <value>
+        /// The <c>_size</c> field represents the size of the generic matrix
+        /// </value>
+        private int _size;
+        /// <value>
+        /// The <c>_matrixElements</c> field stores an array of generic types
+        /// </value>
+        private T?[] _matrixElements;
+        /// <value>
+        /// The <c>ElementChanged</c> event delegate for when an element in the matrix is changed.
+        /// </value>
+        public event EventHandler<ElementChangedEventArgs<T>> ElementChanged;
+        /// <summary>
+        /// An indexer of the matrix class that provides the access to specified element in the matrix
+        /// </summary>
+        /// <param name="i"></param>
+        /// <param name="j"></param>
+        /// <returns></returns>
+        /// <exception cref="IndexOutOfRangeException"></exception>
+        public T? this[int i, int j]
+        {
+            get
+            {
+                if (i < 0 && j < 0 || i >= _size && j >= _size)
+                    throw new IndexOutOfRangeException();
+                
+                else return _matrixElements[i];
```

Comment:
j should be used here, this is square matrix, which is regular matrix of size NxN